src/components/teams/TeamsList.js

```diff
@@ -1,96 +1,83 @@
-import { useMemo, useState, Fragment } from "react";
+import { useMemo, useState, Fragment, useEffect } from "react";
 import { useSettings } from "../../hooks/useSettings";
 import { useView } from "../../hooks/useView";
 import { PrimaryButton, SubmitAndCancelButtons } from "../Buttons";
 import { ConfigureHelpDesk } from "../integrations/ConfigureIntegration";
 import { CenteredLayout, PageHeader } from "../Layout";
 import TitleDescriptionLink from "../TitleDescriptionLink";
 import { __, _x } from "@wordpress/i18n";
-/**
- * Show list of teams
- *
- * @returns {JSX.Element}
- */
+import Spinner from "../Spinner";
+import { ToastError } from "../Errors";
+
 const TeamsList = () => {
-  const { settings, removeTeam } = useSettings();
-  //Has user clicked delete, but not confirmed?
+  const { settings, removeTeam, loading } = useSettings();
   const [isDeleting, setIsDeleting] = useState(false);
-  //Track team to delete, if confirmed.
   const [teamDeleting, setTeamDeleting] = useState(null);
   const { setCurrentView, setCurrentTeam } = useView();
   const teams = useMemo(() => settings.teams, [settings]);
   const [modalTeam, setModalTeam] = useState(null);
+  const [errorMessages, setErrorMessages] = useState([]);
+  const [previousLoading, setPreviousLoading] = useState(false);
+
   const shouldShowConfigureButton = (team) => {
     return (
       team.helpdesk &&
       ["helpscout", "freescout"].includes(team.helpdesk[0].toLowerCase())
     );
   };
 
-  /**
-   * Cancel delete process
-   */
-  function cancelDelete() {
+  const cancelDelete = () => {
     setIsDeleting(false);
     setTeamDeleting(null);
-  }
+  };
 
-  /**
-   * Completes the deletion of a team
-   */
-  function completeDelete() {
+  const completeDelete = () => {
     removeTeam(teamDeleting, () => {
       cancelDelete();
     });
-  }
+  };
 
-  /**
-   * Displays the confirmation and stores ID of team to be deleted
-   */
-  function startDelete(teamId) {
+  const startDelete = (teamId) => {
     setIsDeleting(true);
     setTeamDeleting(teamId);
-  }
+  };
 
-  function goToAccessKey(teamId) {
+  const goToAccessKey = (teamId) => {
     setCurrentTeam(teamId);
     setCurrentView("teams/access_key");
-  }
+  };
+
+  useEffect(() => {
+    if (previousLoading && !loading) {
+      const errors = teams.filter((team) => team.status === "error");
+      if (errors.length > 0) {
+        setErrorMessages(
+          errors.map((team) => {
+            return {
+              message: team.message,
+              id: team.id,
+              name:
+                team.name !== ""
+                  ? team.name
+                  : __("Team", "trustedlogin-connector") + " " + team.id,
```

Comment:
Should this use `.replace()` with `Team %s` to be consistent with string translation?